### PR TITLE
TST: DataFrameGroupBy.shift Loses TimeZone in datetime64 columns

### LINKED ISSUE #30134: DataFrameGroupBy.shift Loses TimeZone in datetime64 columns
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import datetime as dt
import pytz

df = pd.DataFrame({'a':[1,1], 'date':[dt.datetime.now(tz=pytz.utc), dt.datetime.now(tz=pytz.utc)]})

df.info()

df.groupby('a').shift(0).info()

```
#### Problem description

Causes loss of timezeone meaning that you cannot subtract, etc.

Type turns to `datetime64[ns]`

This does not happen with `df.shift()`

#### Expected Output
`datetime64[ns, UTC]`

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.4.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 18.7.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.3
numpy            : 1.17.2
pytz             : 2019.2
dateutil         : 2.8.0
pip              : 19.0.3
setuptools       : 40.8.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : 7.8.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by tranctan**
Adding test case for issue GH30134

- [x] closes #30134
- [x] tests added / passed

### CODE REVIEW

File: pandas/tests/dtypes/test_dtypes.py [hunk 1]
```diff
@@ -323,6 +324,13 @@ def test_tz_standardize(self):
         dtype = DatetimeTZDtype("ns", dr[0].tz)
         assert dtype.tz == tz
 
+    def test_dfgroupby_shift_tz(self):
+        # GH 30134
+        df = pd.DataFrame({"a": [1, 1], "date": dt.datetime.now(tz=pytz.utc)})
```
Comment (WillAyd) [inline pandas/tests/dtypes/test_dtypes.py]:
```suggestion
        df = pd.DataFrame({"a": [1, 1], "date": pd.Timestamp.utcnow()})
```

Slightly more idiomatic

File: pandas/tests/dtypes/test_dtypes.py [hunk 2]
```diff
@@ -323,6 +324,13 @@ def test_tz_standardize(self):
...
+        expected = df.dtypes["date"]
+        result = df.groupby("a").shift(0).dtypes["date"]
+        assert expected == result
```
Comment (WillAyd) [inline pandas/tests/dtypes/test_dtypes.py]: Can you build the actual expected series and use `tm.assert_series_equal(result, expected)`? Should see examples of this throughout tests

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for the PR! Looks good just a few edits

Can you also move this to pandas/tests/groupby/test_transform.py?

File: pandas/tests/groupby/test_transform.py [hunk 3]
```diff
@@ -1138,3 +1138,12 @@ def func(grp):
 
     expected = pd.DataFrame([2, -2, 2, 4], columns=["B"])
     tm.assert_frame_equal(result, expected)
+
+
```
Comment (jreback) [inline pandas/tests/groupby/test_transform.py]: can you move to pandas/tests/groupby/test_groupby.py, near the other _shift tests

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm ping on green

**Comment by simonjayhawkins** [comment]
Thanks @tranctan

**Pull request merged**